src/coreclr/src/tools/aot/ILCompiler.TypeSystem.ReadyToRun.Tests/TestTypeSystemContext.cs

```diff
@@ -57,21 +59,9 @@ public override ModuleDesc ResolveAssembly(System.Reflection.AssemblyName name,
 
         public override FieldLayoutAlgorithm GetLayoutAlgorithmForType(DefType type)
         {
-            if (type == UniversalCanonType)
-                return UniversalCanonLayoutAlgorithm.Instance;
-
             return _metadataFieldLayout;
         }
 
-        protected override RuntimeInterfacesAlgorithm GetRuntimeInterfacesAlgorithmForNonPointerArrayType(ArrayType type)
```

Comment:
Put these under `#if !READYTORUN` instead of deleting them?

Reply:
We could also just pull in the `ArrayOfTRuntimeInterfacesAlgorithm.cs` file from the CoreRT repo, include it in the test project (no need to include from the shipping binary, I think), and it would let us run a couple more tests around array casting.

The algorithm and `Array<T>` is really just a way for the type system to learn what interfaces array has - it's not that important. What matters is that things like array casting work and they build on top of that.

Reply:
(And maybe we could do the same for the `UniversalCanonLayoutAlgorithm`)

Reply:
Ported both the `Array<T>` and USG layout and the tests pass 👍